libs/server/AOF/AofProcessor.cs

```diff
@@ -259,40 +259,113 @@ private unsafe bool ReplayOp(byte* entryPtr)
                     ObjectStoreDelete(objectStoreBasicContext, entryPtr);
                     break;
                 case AofEntryType.StoredProcedure:
-                    ref var input = ref Unsafe.AsRef<SpanByte>(entryPtr + sizeof(AofHeader));
-                    respServerSession.RunTransactionProc(header.type, new ArgSlice(ref input), ref output);
+                    RunStoredProc(header.type, entryPtr);
                     break;
                 default:
                     throw new GarnetException($"Unknown AOF header operation type {header.opType}");
             }
             return true;
         }
 
-        static unsafe void StoreUpsert(BasicContext<SpanByte, SpanByte, SpanByte, SpanByteAndMemory, long, MainSessionFunctions, MainStoreFunctions, MainStoreAllocator> basicContext, byte* ptr)
+        unsafe void RunStoredProc(byte id, byte* ptr)
         {
-            ref var key = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader));
-            ref var input = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader) + key.TotalSize);
-            ref var value = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader) + key.TotalSize + input.TotalSize);
+            var curr = ptr;
+            var parseStateCount = *(int*)curr;
+            curr += sizeof(int);
+
+            var parseState = new SessionParseState();
+            if (parseStateCount > 0)
+            {
+                ArgSlice[] parseStateBuffer = default;
+                parseState.Initialize(ref parseStateBuffer, parseStateCount);
+
+                for (var i = 0; i < parseStateCount; i++)
+                {
+                    ref var sbArgument = ref Unsafe.AsRef<SpanByte>(curr);
+                    parseStateBuffer[i] = new ArgSlice(ref sbArgument);
+                    curr += sbArgument.TotalSize;
+                }
+            }
+
+            respServerSession.RunTransactionProc(id, ref parseState, ref output);
+        }
+
+        static unsafe void StoreUpsert(BasicContext<SpanByte, SpanByte, RawStringInput, SpanByteAndMemory, long, MainSessionFunctions, MainStoreFunctions, MainStoreAllocator> basicContext, byte* ptr)
+        {
+            var curr = ptr + sizeof(AofHeader);
+            ref var key = ref Unsafe.AsRef<SpanByte>(curr);
+            curr += key.TotalSize;
+
+            // Reconstructing RawStringInput
+
+            // input
+            ref var sbInput = ref Unsafe.AsRef<SpanByte>(curr);
+            ref var input = ref Unsafe.AsRef<RawStringInput>(sbInput.ToPointer());
```

Comment:
Its safer to just read the count, and create the RawStringInput and ObjectInput structs on the stack, as they need to be populated with parse state anyway.